### DISCUSSION TITLE
How does the Synapse server consume events?

### DISCUSSION THREAD

**Discussion by mkane91301** [serverlessworkflow/synapse#318] [Q&A] [answered]
Maybe I'm a bit dense and I just didn't catch where this is covered in the documentation, but how do the Cloud Events actually get into the Synapse server? If I have an event publisher do I need an event broker that's capable of calling a gRPC endpoint on the Synapse server? Or is there some configuration that enables the Synapse server to listen on an event broker?

**Comment by JBBianchi** [comment, marked as answer]
Hello @mkane91301 
Thanks for your question, that matter does indeed lack a little documentation.

It's actually very simple. Any `POST` request with a `Content-Type: application/cloudevents+json` header made on the HTTP endpoint (default port 42286) will be processed as a Cloud Event.

**Reply by cdavernas** [reply]
I'd add that the event can be posted on any path, as Synapse will ignore path and query when it pick ups a request with `application/cloudevents+json` content type. Using a specific path can be usefull for log management (i.e. we usually use `/events/pub`, so that we can quickly find and aggregated all CE request by searching for said path in Prometheus)

**Reply by cdavernas** [reply]
A new release will occur today or in the upcoming days, giving the ability to create, trigger, edit and delete correlations, so that you can play with cloud events in app. If you want to have a sneak peak at it, see https://github.com/neuroglia-io/synapse/tree/fix-correlations

Btw, if you like the project, don't hesitate to drop a star as a sign of support ;)

**Comment by ElanHasson** [comment]
Along the same lines, how are events published from workflows? where do they go and how?

**Reply by Flex-Xuan** [reply]
Same questions, after checked the codes, are your publish the event to webhook site?

**Reply by cdavernas** [reply]
They are published to the configured `CloudEventBusOptions.BrokerUri`.
You can set it using `services.Configure<CloudEventBusOptions>()`.
You can find the related code at:
https://github.com/neuroglia-io/framework/tree/v2.1.4/src/Eventing/Neuroglia.Eventing.CloudEvents

**Reply by cdavernas** [reply]
Sorry @ElanHasson, I had overlooked your last question 😞

**Comment by Flex-Xuan** [comment]
> Hello @mkane91301 Thanks for your question, that matter does indeed lack a little documentation.
> 
> It's actually very simple. Any `POST` request with a `Content-Type: application/cloudevents+json` header made on the HTTP endpoint (default port 42286) will be processed as a Cloud Event.

Can you show me where this was handled in the codes, because when I host Synapse on my laptop for researching, although I post the request with content-type you said, but nothing happens.

**Reply by cdavernas** [reply]
https://github.com/neuroglia-io/framework/blob/v2.1.4/src/Eventing/Neuroglia.Eventing.CloudEvents.AspNetCore/CloudEventMiddleware.cs

**Reply by Flex-Xuan** [reply]
I see, Thanks a lot !
